src/s390_ecc.c

```diff
@@ -1743,16 +1743,17 @@ struct {				\
 #ifdef ICA_FIPS
 			if (fips & ICA_FIPS_MODE) {
 				fc |= 0x80; /* deterministic signature */
-				RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+				do {
+					RAND_bytes(param.P256.rand + off, sizeof(param.P256.rand) - off);
+					rc = s390_kdsa(fc, param.buff, NULL, 0);
+				while (rc)
```

Comment:
You don't want to do below s390_kdsa() again if you exit out of this loop. 
I guess the below s390_kdsa() needs to be on the else case of the above if.

Reply:
same at the other 2 places for the other curves of course.

Reply:
yes, right ...